src/storage/page_buffer.c

```diff
@@ -7735,29 +7732,9 @@ pgbuf_claim_bcb_for_fix (THREAD_ENTRY * thread_p, const VPID * vpid, PAGE_FETCH_
       perfmon_inc_stat (thread_p, PSTAT_PB_NUM_IOREADS);
       show_status->num_pages_read++;
 
-#if defined(ENABLE_SYSTEMTAP)
-      query_id = qmgr_get_current_query_id (thread_p);
-      if (query_id != NULL_QUERY_ID)
-	{
-	  monitored = true;
-	  CUBRID_IO_READ_START (query_id);
-	}
-#endif /* ENABLE_SYSTEMTAP */
+      int error_code = pgbuf_read_page_from_file_or_page_server (thread_p, vpid, &bufptr->iopage_buffer->iopage);
```

Comment:
CUBRID_IO_READ_START should be registered before reading the page, CUBRID_IO_READ_END after. Last argument of CUBRID_IO_READ_END is 1 if error occurred and 0 if it was successful.
```suggestion
#if defined(ENABLE_SYSTEMTAP)
      bool monitored = false;
      QUERY_ID query_id = qmgr_get_current_query_id (thread_p);
      if (query_id != NULL_QUERY_ID)
	{
	  monitored = true;
	  CUBRID_IO_READ_START (query_id);
	}
#endif /* ENABLE_SYSTEMTAP */

      int error_code = pgbuf_read_page_from_file_or_page_server (thread_p, vpid, &bufptr->iopage_buffer->iopage);
      
#if defined(ENABLE_SYSTEMTAP)
      if (monitored)
        {
          CUBRID_IO_READ_END (query_id, IO_PAGESIZE, error_code != NO_ERROR);
        }
#endif  /* ENABLE_SYSTEMTAP */
```